src/gui/transferlistmodel.cpp

```diff
@@ -184,7 +184,7 @@ QVariant TransferListModel::data(const QModelIndex &index, const int role) const
     case TR_PROGRESS:
         return torrent->progress();
     case TR_STATUS:
-        return QVariant::fromValue(torrent->state());
+        return (role == Qt::DisplayRole) ? QVariant::fromValue(torrent->state()) : torrent->error();
```

Comment:
It seems that it is wrong that the Model returns some codes, etc. (instead of displayable value) in DisplayRole context.
This remark applies not only to your changes, but also to the existing code. Unfortunately, GUI code is still mostly unclean...

Reply:
I find it a bit hard to comply with the ideal ItemDataRole spec. A quick look seems that I need to pass `*torrent` in `Qt::UserRole` and do the work in delegate and sortModel.

Reply:
>I need to pass torrent pointer in Qt::UserRole and do the work in delegate and sortModel.

Ideally you should pass strings as DisplayRole. But if strings are not suitable for sorting and/or other purposes, then you should return the appropriate data as other (probably user-defined) roles. E.g. you can return raw data as UserRole.
But as I said before it's applied to existing code mostly so you can ignore it in this PR.